### ISSUE TITLE
urxvt shows transparent images

### ISSUE THREAD

**Issue by catzybluphish** [ranger/ranger#1416]
- Operating system and version: Arch (everything is the latest)
- Terminal emulator and version:  urxvt
- Python version: 
- Ranger version/commit: 
- Locale: en_US

#### Current Behavior
If I preview an image, it'll appear semi transparent:
![sozlh9](https://user-images.githubusercontent.com/40843060/50373802-a8660a80-062f-11e9-8b0d-00d69e5869c8.png)

#### Expected Behavior
All images to appear opaque in urxvt.
#### Context
I'm running urxvt with true transparency with compton but the problem goes away if I use fake transparency. It'll always appear semi transparent for images that are rendered but sometimes for those that aren't, it'll also seem semi transparent:
![2018-12-22-213306_680x424_scrot](https://user-images.githubusercontent.com/40843060/50373907-45757300-0631-11e9-9de9-d744989930d4.png)
^not rendered
This may not be exclusively ranger's fault because if I set an image for neofetch, it shows the same semi transparency but just wanted to know if you guys have any clues to the problem of this.

[ranger configs](https://github.com/catzybluphish/Arch-dotfiles/tree/master/.config/ranger)
[.Xresources](https://github.com/catzybluphish/Arch-dotfiles/blob/master/.Xresources)

**Comment by toonn** [comment]
Are you using a transparent background? Cause the urxvt method just sets the background basically, so it necessarily uses the same transparency as your background.

**Comment by catzybluphish** [comment]
> Are you using a transparent background? Cause the urxvt method just sets the background basically, so it necessarily uses the same transparency as your background.

None the of images and backgrounds are transparent except the outside of the rendered man. I'm using w3m as my urxvt doesn't have pixbuf support.

**Comment by GloverDonovan** [comment]
Back when I used urxvt, I had all sorts of problems with transparency and images in the terminal. The whole thing is basically a mess and not worth your time.

I recommend using a modern terminal emulator like [kitty](https://github.com/kovidgoyal/kitty) instead. Images will render as expected, and there are several ways to change opacity on the fly.

**Comment by catzybluphish** [comment]
Yes, using true transparency.

**Comment by catzybluphish** [comment]
> @catzybluphish, to be clear, I'm talking about your terminals background. Is you terminal transparent?

Yes, using true transparency.

**Comment by Vifon** [comment]
The images is rendered inside the terminal window's buffer so it is affected by what compton does to the window as a whole.

**Comment by catzybluphish** [comment]
> The images are rendered inside the terminal window's buffer so it is affected by what compton does to the window as a whole.

Yeah, it must be on compton's end. But how come some images appear completely opaque like how they're supposed to but others (like shown above) do not? I've heard w3m just isn't reliable in general...

**Comment by Vifon** [comment]
Wait, *some images* aren't transparent? Well, this is getting better and better. Sadly, not much we can do on our end.

**Comment by ntguardian** [comment]
I was able to replicate this problem and discovered why some images are transparent and others are not: the transparent images are PNG files. The non-transparent images were JPEG, which is a file format that does not allow transparency. PDF previews for me also don't have transparency problems; since they're temporarily rendered they probably were rendered in a way that doesn't allow for transparency.

Is it possible to have ranger render all images the same way as it renders JPEGs and whatever it's using to generate PDFs? Sorry, just started using ranger a few minutes ago.

**Comment by toonn** [comment]
Hmm, let's keep this open. I'm still not quite clear on the issue though. Could someone whip up a *simple* reproducible example? Like create a simple image with inkscape that has some transparent and opaque areas and export as both jpeg and png. *And* clearly explain what exactly is going wrong and what you expect, and everything about the terminal/WM/preview method/anything that might be relevant.

PDFs are probably just rendered as jpeg so if that doesn't support transparency (does it really not?) that's where the opaqueness is coming from.

**Comment by marcospb19** [comment]
## Tests
Ranger is using w3m. \
compton `--config /dev/null`


I tested .png and .jpg images with two terminal emulators:
> termite (true transparency) \
> urxvt (fake and true transparency)

#### Example test:

The .png file have transparency (i also tested .png files without transparency/alpha channel variations):

![2019-06-16-213918_226x226_scrot](https://user-images.githubusercontent.com/38900226/59577057-6aa2e580-9098-11e9-80d3-a183f7f801d8.png)

This picture shows the first test i did, *termite* with `true` on the __left__, and *urxvt* with `fake` on the __right__:

![2019-06-16-183951_1577x705_scrot](https://user-images.githubusercontent.com/38900226/59576933-e51f3580-9097-11e9-9538-c3ac01fb8579.png)

And then i tested the urxvt with true transparency as described.

# Results:
When a .png file is shown inside a true transparency terminal, the image content becomes transparent as well as the background.

### Expected behavior
Image stays opaque like the TEXT on the terminal.

### Actual behavior
Image became transparent like the terminal BACKGROUND

| Terminal Emulator | Transparency | File Format | Issue |
| ----------------- | ------------ | ----------- | ----- |
| URxvt             | True         | .png        | Yes   |
| URxvt             | True         | .jpg        | No    |
|                   |              |             |       |
| URxvt             | False        | .png        | No    |
| URxvt             | False        | .jpg        | No    |
|                   |              |             |       |
| Termite           | True         | .png        | Yes   |
| Termite           | True         | .jpg        | No    |


## Conclusion:
If this is a bad usage of the w3m program (any flag missing?) then this is a ranger issue.

If this is a w3m limitation, the it is a w3m issue, and it's harder to fix because w3m is not hosted on github.

I think that this is a w3m problem.
